collect_app/src/main/java/org/odk/collect/android/geo/OsmDroidMapFragment.java

```diff
@@ -374,6 +361,31 @@
 
     @Override public void onClientStop() { }
 
+    private void enableLocationUpdates(boolean enable) {
+        if (locationClient == null) {
+            locationClient = LocationClients.clientForContext(getActivity());
+            locationClient.setListener(this);
+        }
+        if (enable) {
+            LocationManager locationManager = (LocationManager) getContext().getSystemService(Context.LOCATION_SERVICE);
+            if (locationManager != null && locationManager.isProviderEnabled(LocationManager.GPS_PROVIDER)) {
```

Comment:
I'm not quite understanding your comment about #2997. This is what causes the discrepancy, right? I think the `enableLocationUpdates` method here could be identical to the one in the GoogleMapsFragment. I'm happy for that to happen now.

Reply:
Oh, I just meant to connect the issues because I suspect it will make sense to address them together.  Right now Mapbox is the odd one out with its own LocationEngine infrastructure — we'll probably want to unify them all.  (Fortunately, Mapbox provided hooks so we can easily disable their LocationEngine and use our own instead.)

Reply:
The first part can be identical to the one in GoogleMapFragment, but this one needs extra stuff to enable and disable the location overlay as well.   Am I maybe misunderstanding what you have in mind?

Reply:
Ahh, got it. I'm saying that modulo the overlay, this work to check whether the GPS provider is enabled and to show an error otherwise is what results in the discrepancy.